### PR TITLE
Add docs on allpication modules and operating modes

### PR DESCRIPTION
**Pull request by AlexeyKasatkin**
Closes: #100

### CODE REVIEW

File: doc/operating-modes.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# IPs distribution in Daemon Set mode
+
+There can be different rules of IPs distribution among controllers (i.e. nodes).
+This is controlled by `nodefilter` parameter. Default rule is to distribute IPs
+evenly among all the controllers (`nodefilter=fair`). Alternative rule is
+`nodefilter=first-alive` where all IPs will be spawned on the first available
+controller (i.e. node). Daemon Set mode with `first-alive` rule is similar to
+Simple mode but with more responsive and correct fail-over.
+
+
```
Comment (adidenko) [inline doc/operating-modes.md]: Extra lines

File: doc/operating-modes.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Kubernetes provides fail-over for External IP controller application. So, when
+there's a problem with k8s node, External IP controller will be spawned on
+another k8s worker node and will bring External IPs up on that node.
+Simple mode is easy to setup and takes less resources. It makes sense when all
+IPs should be brought up on the same node. However, fail-over in this mode takes
+longer than in daemon set mode (k8s detects node failure in much longer
+intervals by default, see `node-monitor-grace-period`, `pod-eviction-timeout`;
+these could be minimized but there was no research on how safe is it to set them
+to some 2-5 seconds) and it may work wrong in some cases.
+
```
Comment (adidenko) [inline doc/operating-modes.md]: Please add full command line including arguments showing how to start application in such mode.

File: doc/operating-modes.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+External IP controller application will be run on several nodes. One or more
+controller modules and one or more scheduler modules will be run. Controller
+modules should be run on nodes where IPs are expected to be spawned. Scheduler
+modules can be run on any nodes. There is no much sense to run more than one
+scheduler on every particular node. Several scheduler modules are run to provide
+HA for scheduler (A/B mode). So, that in case of no response from active
+scheduler (e.g. on node failure) another one becomes active. If more than one
+scheduler will be used then scheduler election mode should be switched on
+(parameter `leader-elect=true`).
+
```
Comment (adidenko) [inline doc/operating-modes.md]: Same here. Please add full command line including arguments showing how to start application in such mode.

File: doc/operating-modes.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+controllers (i.e. nodes).
+Scheduler(s) can be run on any nodes as they just schedule claims among the
+controllers. Scheduler module is not obligatory to be run. It is not in use
+while application runs in Simple mode.
+
+## Operating Modes
+
+External IP Controller application may be run in one of the operating modes:
+* [Simple](#simple-mode)
+* [Daemon Set](#daemon-set)
```
Comment (dshulyak) [inline doc/operating-modes.md]: Claims mode

File: doc/operating-modes.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+External IP controller application will be run on several nodes. One or more
+controller modules and one or more scheduler modules will be run. Controller
+modules should be run on nodes where IPs are expected to be spawned. Scheduler
+modules can be run on any nodes. There is no much sense to run more than one
+scheduler on every particular node. Several scheduler modules are run to provide
+HA for scheduler (A/B mode). So, that in case of no response from active
+scheduler (e.g. on node failure) another one becomes active. If more than one
+scheduler will be used then scheduler election mode should be switched on
+(parameter `leader-elect=true`).
```
Comment (dshulyak) [inline doc/operating-modes.md]: makes sense to add that preferable scheduler shoudn't be run on same node as controller, cause in the event of node outage - ip failover time will increase.

File: doc/operating-modes.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+modules should be run on nodes where IPs are expected to be spawned. Scheduler
+modules can be run on any nodes. There is no much sense to run more than one
+scheduler on every particular node. Several scheduler modules are run to provide
+HA for scheduler (A/B mode). So, that in case of no response from active
+scheduler (e.g. on node failure) another one becomes active. If more than one
+scheduler will be used then scheduler election mode should be switched on
+(parameter `leader-elect=true`) otherwise there can be race condition between
+schedulers.
+It is better not to run scheduler on same node as controller because in case of
+node outage IP fail-over will take more time.
```
Comment (dshulyak) [inline doc/operating-modes.md]: maybe re-schedulling would fit better

File: doc/operating-modes.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+External IP controller application will be run on several nodes. One or more
+controller modules and one or more scheduler modules will be run. Controller
+modules should be run on nodes where IPs are expected to be spawned. Scheduler
+modules can be run on any nodes. There is no much sense to run more than one
+scheduler on every particular node. Several scheduler modules are run to provide
+HA for scheduler (A/B mode). So, that in case of no response from active
+scheduler (e.g. on node failure) another one becomes active. If more than one
+scheduler will be used then scheduler election mode should be switched on
+(parameter `leader-elect=true`) otherwise there can be race condition between
```
Comment (dshulyak) [inline doc/operating-modes.md]: multiple conditions sounds better

File: doc/operating-modes.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+Alternative rule is `nodefilter=first-alive` where all IPs will be spawned on
+the first available controller (i.e. node). Claims mode with `first-alive`
+rule is similar to Simple mode but with more responsive and correct fail-over.
+
+# Parameters
+
+Next command-line parameters are available in Claims mode for controller module:
+* `iface` - interface that will be used to assign IP addresses (default "eth0").
+* `hb` - how often to send heartbeats from controllers (default 2 sec).
+* `kubeconfig` - kubeconfig to use with kubernetes client (default "").
```
Comment (dshulyak) [inline doc/operating-modes.md]: we should specify that if it wont be provided - incluster configuration for auth will be used

File: doc/operating-modes.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+* `mask` - mask part of network CIDR (default "32").
+* `resync` - interval to resync state for all ips (default 20 sec).
+* `hostname` - use provided hostname instead of os.Hostname (default
+os.Hostname).
+
+Next command-line parameters are available in Claims mode for scheduler module:
+* `kubeconfig` - kubeconfig to use with kubernetes client (default "").
+* `mask` - mask part of network CIDR (default "32").
+*	`nodefilter` - node filter to use while dispatching IP claims; it controls IPs
```
Comment (dshulyak) [inline doc/operating-modes.md]: removes spaces?

File: doc/operating-modes.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+the first available controller (i.e. node). Claims mode with `first-alive`
+rule is similar to Simple mode but with more responsive and correct fail-over.
+
+# Parameters
+
+Next command-line parameters are available in Claims mode for controller module:
+* `iface` - interface that will be used to assign IP addresses (default "eth0").
+* `hb` - how often to send heartbeats from controllers (default 2 sec).
+* `kubeconfig` - kubeconfig to use with kubernetes client (default "").
+* `mask` - mask part of network CIDR (default "32").
```
Comment (dshulyak) [inline doc/operating-modes.md]: with auto-allocation it won't be needed

File: doc/operating-modes.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+longer than in Claims mode (k8s detects node failure in much longer intervals by
+default, see `node-monitor-grace-period`, `pod-eviction-timeout`;
+these could be minimized but there was no research on how safe is it to set them
+to some 2-5 seconds) and it may work wrong in some cases.
+
+# Parameters
+
+Next command-line parameters are available in Simple mode for controller module:
+* `iface` - interface that will be used to assign IP addresses (default "eth0").
+* `kubeconfig` - kubeconfig to use with kubernetes client (default "").
```
Comment (dshulyak) [inline doc/operating-modes.md]: same as below

**Pull request merged**